src/map/battlefield.cpp

```diff
@@ -307,7 +308,7 @@ bool CBattlefield::InsertEntity(CBaseEntity* PEntity, bool enter, BATTLEFIELDMOB
         PEntity->loc.zone->PushPacket(PEntity, CHAR_INRANGE, new CEntityUpdatePacket(PEntity, ENTITY_SPAWN, UPDATE_ALL_MOB));
         m_NpcList.push_back(static_cast<CNpcEntity*>(PEntity));
     }
-    else if (PEntity->objtype == TYPE_MOB || PEntity->objtype == TYPE_PET)
+    else if (PEntity->objtype == TYPE_MOB || PEntity->objtype == TYPE_PET || PEntity->objtype == TYPE_TRUST)
```

Comment:
Is the intent to treat trusts like battlefield allies? Right now I think allies are only instanced mobs that help during the fight, like Prishe in the Promathia fight. Player pets aren't stored in the battlefield object at all. I would probably remove both of these additions and change [this line](https://github.com/project-topaz/topaz/blob/7bc656841594711c48f8bbef80e3f66cbc7670ee/src/map/battlefield.cpp#L392) to handle trusts as well.